src/schema/meta/context.yaml

```diff
@@ -187,7 +187,15 @@ context:
             n_cols:
               description: 'Number of columns in bvec file'
               type: integer
-
+        nirs:
+          description: 'contents of _nirs.json used by checks for channels.tsv'
+          type: object
+        channels:
+          type: boolean
+          description: 'boolean if _channels.tsv exists'
+        coordsystem:
+          type: boolean
+          description: 'boolean if _coordsystem.json exists'
```

Comment:
This and the comment on `checks/nirs.yaml` are a joint proposal for how to encode these rules that span files. Instead of making checks on a `channels.tsv` file and lookup up an associated `nirs.json`, my proposal puts the rule on `_nirs.snirf`, retrieves the metadata through the `sidecar` context object, and makes assertions on the contents of `associations.channels`. This is where we would define the properties of `associations.channels`. I also expanded `coordsystem` to be more like other associations that contain a path value.

Reply:
Thank you